.github/workflows/ci.yaml

```diff
@@ -41,3 +41,35 @@ jobs:
       - name: Build Hyprland with LEGACY_RENDERER
         run: |
           make legacyrenderer
+
+  meson:
+    name: "Build Hyprland with Meson (Arch)"
+    runs-on: ubuntu-latest
+    container:
+      image: archlinux
+    steps:
+      - name: Download dependencies 📥
+        run: |
+          sed -i 's/SigLevel    = Required DatabaseOptional/SigLevel    = Optional TrustAll/' /etc/pacman.conf
+          pacman --noconfirm --noprogressbar -Syyu
+          pacman --noconfirm --noprogressbar -Sy glslang libepoxy libfontenc libxcvt libxfont2 libxkbfile vulkan-headers vulkan-validation-layers xcb-util-errors xcb-util-renderutil xcb-util-wm xorg-fonts-encodings xorg-server-common xorg-setxkbmap xorg-xkbcomp xorg-xwayland git go clang lld libc++ pkgconf meson ninja wayland wayland-protocols libinput libxkbcommon pixman glm libdrm libglvnd cairo pango systemd scdoc base-devel seatd
+      - name: Checkout Hyprland 📡
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Configure 🔧
+        run: |
+          meson obj-x86_64-pc-linux-gnu \
+            -Ddefault_library=static
+      - name: Compile 🎲
+        run: ninja -C obj-x86_64-pc-linux-gnu
+      - name: Compress artifacts 📦
+        run: |
+          mkdir x86_64-pc-linux-gnu
+          DESTDIR=$PWD/x86_64-pc-linux-gnu meson install -C obj-x86_64-pc-linux-gnu --tags runtime
+          tar -cvf x86_64-pc-linux-gnu.tar.xz x86_64-pc-linux-gnu
```

Comment:
???

Reply:
build artifacts

Reply:
why and what for

Reply:
wont be, binary packages are made from releases

Reply:
> wont be, binary packages are made from releases

well, for nightly binary packages for every commit and maybe you not having to build an archive for each release and instead have a cloud-built release that might be somewhat reproducible and doesn't leak data from your personal system.

Reply:
if you want that, then having it make a full "release" type zip would be better. That means including the wlroots .so file, binary, examples/, hyprctl, assets/ and LICENSE.

Reply:
I've considered doing that, but that doesn't fit the scope of "artifact" anymore. This just contains what is necessary to run Hyprland, so Hyprland itself (no need for wlroots, due to it being statically linked), hyprctl the desktop file and the sample config.

Reply:
> If you do not do that, it's useless. for proper hyprland launching, assets/ is also required

then it should be included then. I was just unaware

> furthermore, I do not really care that you link statically, I'd rather you not, as everything so far is made without the static link in mind (CMake and packages)

this obv. doesn't affect existing packages and I ensured cmake still works. That doesn't change the fact of the matter, that this is one of the few cases where static linking makes sense, since it's the best way to avoid conflicts with os-provided libwlroots that doesn't involve shipping wlroots itself.

Reply:
thats why the wlroots ver is 11032. It will cause issues, hyprland-bin will be broken without an so. I know it makes sense, but we have worked with dynamic linking for a while now and making 2 build systems that do NOT produce the exact same result will be a pain.

Reply:
> and this way it causes also issues for anything that isn't hyprland but links against wlroots.

wrong. They will link against libwlroots.0.X.so and not libwlroots.0.11032.so

> since you built wlroots always from source anyways, why not link against it statically too, if consistency is so important to you?

can CMake do that? if you can make it like this here, and make Kainoa accept it ( @ThatOneCalculator ) then I can accept that solution. But then I'd like you to somehow (if possible) force static linking, so it's not an --option

Reply:
> wrong. They will link against libwlroots.0.X.so and not libwlroots.0.11032.so

but when linking, you only specify `-lwlroots`, how does ld figure out which version to use?

> can CMake do that? if you can make it like this here, and make Kainoa accept it ( @ThatOneCalculator ) then I can accept that solution

I despise cmake syntax, but I'll take a look

Reply:
> but when linking, you only specify -lwlroots, how does ld figure out which version to use?

![magiccover.jpg](https://user-images.githubusercontent.com/43317083/173595037-690b9af9-ad27-4afa-9b26-2dce16499f0f.jpg)

magic

> I despise cmake syntax, but I'll take a look

thank you :-)

Reply:
okay, then it seems like I'd have to do some major hackage to make it work. We can leave it at that then, and just make @ThatOneCalculator make the package ignore libwlroots if not present, JIC. I will probably still be publishing releases from CMake.

If we want a nightly build though, please consult with Kainoa what you need to include in the tar for a package.

Reply:
>> but when linking, you only specify -lwlroots, how does ld figure out which version to use?
>
> magiccover.jpg
>
> magic

well, it looks like it just overwrites the libwlroots.so symlink that it points towards libwlroots.so.11032. it also overwrites wlroots header and the wlroots.pc pkg-config manifest. this will cause issues for non-hyprland uses down the road for most users I imagine.